app/controllers/finders_controller.rb

```diff
@@ -37,4 +38,8 @@ def facet_params
   def keywords
     params[:keywords] unless params[:keywords].blank?
   end
+
+  def atom_url
+    request.original_url.gsub('.json', '').gsub(finder_slug, "#{finder_slug}.atom")
```

Comment:
JSON issues aside, i think the controller method could be a lot cleaner, but i'd settle for some tests around it so it can be cleaned up later.